### PR TITLE
Add task notes step to DAG docs tutorial

### LINKED ISSUE #1902: Add guidance on using DAG and task run comments
Airflow 2.5 introduced https://github.com/apache/airflow/pull/26457, which lets you write comments for DAG or task runs.

I was thinking either https://docs.astronomer.io/learn/airflow-ui or https://docs.astronomer.io/learn/custom-airflow-ui-docs-tutorial would be a good place to include this information. We could provide more guidance on what the feature should be used for and how to use it - I don't think there are any docs currently on this!

### PR DESCRIPTION
**Pull request by kentdanas**
Resolves #1902 - we have docs on task/DAG run notes in our rerunning DAGs guide, but several folks have had trouble finding that. Adding this step to our DAG docs tutorial will hopefully be another useful place to document this feature.

### CODE REVIEW

File: learn/custom-airflow-ui-docs-tutorial.md [hunk 1]
```diff
@@ -325,6 +325,26 @@ tell_me_what_to_do = PythonOperator(
 
     ![All Task Docs](/img/guides/task_docs_all.png)
 
+## Step 5: Add notes to a task instance and DAG run
+
+Another way of adding information to your DAGs and tasks is to add notes to task instances and DAG runs from the Grid View in the Airflow UI. This feature is useful if you need to add information specific to a single run of your DAG. 
+
+1. Go to the **Grid View** of the `docs_example_dag.py` DAG you created in [Step 2](#step-2-create-a-new-dag).
```
Comment (TJaniF) [inline learn/custom-airflow-ui-docs-tutorial.md:332]:
```suggestion
1. Go to the **Grid View** of the `docs_example_dag` DAG you created in [Step 2](#step-2-create-a-new-dag).
```

**Review by TJaniF — APPROVED** [review]
👍 One tiny suggestion but LGTM. :)

**Comment by kentdanas** [comment]
Thanks @TJaniF! @jwitz want to take a quick look at the copy on this one before I merge?

File: learn/custom-airflow-ui-docs-tutorial.md [hunk 2]
```diff
@@ -325,6 +325,26 @@ tell_me_what_to_do = PythonOperator(
 
     ![All Task Docs](/img/guides/task_docs_all.png)
 
+## Step 5: Add notes to a task instance and DAG run
+
+Another way of adding information to your DAGs and tasks is to add notes to task instances and DAG runs from the Grid View in the Airflow UI. This feature is useful if you need to add information specific to a single run of your DAG.
```
Comment (jwitz) [inline learn/custom-airflow-ui-docs-tutorial.md:330]:
```suggestion
Starting in Airflow 2.5, you can add notes to task instances and DAG runs from the **Grid** view in the Airflow UI. This feature is useful if you need to share contextual information about a DAG or task run with your team, such as why a specific run failed.
```

File: learn/custom-airflow-ui-docs-tutorial.md [hunk 3]
```diff
@@ -325,6 +325,26 @@ tell_me_what_to_do = PythonOperator(
...
+
+3. Click **Details** > **Task Instance Notes** or **DAG Run notes** > **Add Note**.
+
+4. Write a note and click **Save Note**.
+
+ ![Add task note](/img/guides/2_5_task_notes.png)
+
+:::note
+
+This feature is only available in Airflow version 2.5+. Ensure your Dockerfile is using at least version 7.0 of Astro runtime to complete this step.
```
Comment (jwitz) [inline learn/custom-airflow-ui-docs-tutorial.md:344]:
```suggestion
This feature is only available in Airflow version 2.5+. Ensure your Dockerfile is using at least version 7.0 of Astro Runtime to complete this step.
```

Comment (jwitz) [reply learn/custom-airflow-ui-docs-tutorial.md:344]: Im not sure that we need this note (don't we always assume the latest version for maintainence sake?), but if you want to keep it then let's move it to the top of the step or into the prerequisites.

**Review by jwitz — APPROVED** [review]
LGTM! Thanks for getting this added :)

**Pull request merged**